File: Scala/currying/2/main.scala
/**
  * Created by fay on 2017/6/1.
  */

object main {
  def main(args: Array[String]): Unit = {
    add(8)(10)
  }

  def add(x: Int) = (y: Int) => {
    println(x + y)
  }
}

File: Java/currying/2/main.java
/**
 * Created by fay on 2017/5/9.
 */

public class main {
    public static void main(String args[]) {
        main m = new main();
        System.out.println(m.add(8).add(10));
    }

    lambda add(int x) {
       return y -> {
           return x + y;
       };
    }
}

interface lambda {
    int add(int y);
}
